src/common/impl_registration.hpp

```diff
@@ -203,13 +203,13 @@
 
 // Primitive CPU ISA section is in src/cpu/platform.hpp
 
-#if BUILD_PRIMITIVE_GPU_ISA_ALL || BUILD_GEN9
+#if BUILD_PRIMITIVE_GPU_ISA_ALL
 #define REG_GEN9_ISA(...) __VA_ARGS__
 #else
 #define REG_GEN9_ISA(...)
 #endif
 
-#if BUILD_PRIMITIVE_GPU_ISA_ALL || BUILD_GEN11
+#if BUILD_PRIMITIVE_GPU_ISA_ALL
 #define REG_GEN11_ISA(...) __VA_ARGS__
 #else
 #define REG_GEN11_ISA(...)
```

Comment:
Probably the easiest way to disable Gen9/11 without any precision surgery. We can handle proper cleanup later.

```suggestion
#define REG_GEN9_ISA(...)
#define REG_GEN11_ISA(...)
```

Reply:
I think it might better to keep this PR for the user-visible changes and have a follow-up PR for the internal changes?
I've preserved some of the suggestions in the mzhukova/main/remove-gen9-11-ext branch and created a followup task https://jira.devtools.intel.com/browse/MFDNN-13645.